File: beekeeping-api/BeekeepingApi/Models/BeehiveComponent.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace BeekeepingApi.Models
{
    [Flags]
    public enum ComponentTypes
    {
        Meduvė = 1,
        Pusmeduvė = 2,
        Išleistuvas = 4,
        SkiriamojiTvorelė = 8,
        Aukštas = 16,
        DugnoSklendė = 32,
        Maitintuvė = 64
    }

    public class BeehiveComponent
    {
        [Key]
        public long Id { get; set; }

        public ComponentTypes Type { get; set; }

        [Range(1, 9)]
        public int? Position { get; set; }

        public DateTime InstallationDate { get; set; }

        public long BeehiveId { get; set; }

        [ForeignKey("BeehiveId")]
        public Beehive Beehive { get; set; }
    }
}

File: beekeeping-app/src/app/_models/beehive-component.ts
export class BeehiveComponent {
    id: number;
    type: BeehiveComponentType;
    position?: number;
    installationDate: Date;
    beehiveId: number;
}

export enum BeehiveComponentType {
    Meduvė = 1, //HoneySuper
    Pusmeduvė = 2, //HoneyMiniSuper
    Išleistuvas = 4, //bee decreaser
    SkiriamojiTvorelė = 8, //queen excluder
    Aukstas = 16, //super
    DugnoSklendė = 32, //bottom Gate
    Maitintuvė = 64 //feeder
}

export const BeehiveComponentType2LabelMapping: Record<BeehiveComponentType, string> = {
    [BeehiveComponentType.Meduvė]: "Meduvė",
    [BeehiveComponentType.Pusmeduvė]: "Pusmeduvė",
    [BeehiveComponentType.Išleistuvas]: "Išleistuvas",
    [BeehiveComponentType.SkiriamojiTvorelė]: "Skiriamoji tvorelė",
    [BeehiveComponentType.Aukstas]: "Aukštas",
    [BeehiveComponentType.DugnoSklendė]: "Dugno sklendė",
    [BeehiveComponentType.Maitintuvė]: "Maitintuvė"
};